src/components/directory-content.jsx

```diff
@@ -487,360 +474,44 @@ const DirectoryContent = () => {
-                .catch((error) => {
-                    if (Object.keys(currentChildrenRef.current).length === 0) {
-                        handleError(error.message);
-                    }
-                })
-                .finally(() => {
-                    // discarding request for older directory
-                    if (currentChildrenRef.current === currentChildren) {
-                        setChildrenMetadata(metadata);
-                        setIsMissingDataAfterDirChange(false);
-                    }
-                });
-        }
-        setSelectedUuids(new Set());
-    }, [handleError, currentChildren, currentChildrenRef]);
+    const isActiveElementUnchecked = useMemo(
+        () =>
+            activeElement &&
+            !checkedRows.find(
+                (children) => children.elementUuid === activeElement.elementUuid
+            ),
+        [activeElement, checkedRows]
+    );
 
-    const getSelectedChildren = () => {
-        let selectedChildren = [];
-        if (currentChildren?.length > 0) {
-            // Adds the previously selected elements
-            if (selectedUuids?.size) {
-                selectedChildren = currentChildren
-                    .filter(
-                        (child) =>
-                            selectedUuids.has(child.elementUuid) &&
-                            child.elementUuid !== activeElement?.elementUuid
-                    )
-                    .map((child) => {
-                        return {
-                            subtype:
-                                childrenMetadata[child.elementUuid]?.subtype,
-                            hasMetadata:
-                                childrenMetadata[child.elementUuid] !==
-                                undefined,
-                            ...child,
-                        };
-                    });
-            }
+    const handleRowSelected = useCallback(() => {
+        setCheckedRows(computeCheckedElements(gridRef, childrenMetadata));
+    }, [childrenMetadata]);
 
-            // Adds the active element
-            if (activeElement) {
-                selectedChildren.push({
-                    ...activeElement,
-                    subtype:
-                        childrenMetadata[activeElement.elementUuid]?.subtype,
-                    hasMetadata:
-                        childrenMetadata[activeElement.elementUuid] !==
-                        undefined,
-                });
-            }
+    //It includes checked rows and the row with its context menu open
+    const fullSelection = useMemo(() => {
+        const selection = [...checkedRows];
+        if (isActiveElementUnchecked) {
+            selection.push(formatMetadata(activeElement, childrenMetadata));
         }
-        return [...new Set(selectedChildren)];
-    };
+        return selection;
+    }, [
+        activeElement,
+        checkedRows,
+        childrenMetadata,
+        isActiveElementUnchecked,
+    ]);
```

Comment:
I don't think all of this is necessary since you now add active element to selection when right clicking
after a few tests I think you can remove it and isActiveElementUnchecked

This is what I meant when I said it would simplify the code to keep old behaviour with contextMenuPolicy

Reply:
Oh I see, I'll proceed to get rid of this

Reply:
I guess it could be fixed and simplified by a deep refactoring of both the content toolbar and the contextual menu but I'd avoid bulldoze everything for this PR